docs/reference/platforms.rst

```diff
@@ -0,0 +1,215 @@
+
+See :ref:`build-on <reference-build-on>` and
+:ref:`build-for <reference-build-for>` for information on the ``build-on``
+and ``build-for`` keywords.
+
+.. _reference-platforms-shorthand:
+
+Shorthand notation
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <arch>:
+
+Shorthand notation is a simple way to describe charms that build on and build
+for the same architecture. Shorthand notation requires the platform name to be
+a :ref:`valid debian architecture <reference-architectures>`.
+
+.. _reference-platforms-multi-base:
+
+Multi-base notation
+~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on: [<base>:<arch>, <base>:<arch>]
+       build-for: [<base>:<arch>]
+
+Multi-base charms define a :ref:`base <reference-bases>` in each platform entry
+instead of the top-level ``base`` and ``build-base`` keywords. Within each
+platform entry, the base must be the same for all ``build-on`` and
+``build-for`` entries.
+
+The recommended platform name is ``<distribution>-<series>-<build-for-arch>``
+(for example, ``ubuntu-24.04-riscv64``).
+
+.. _reference-platforms-multi-base-shorthand:
+
+Multi-base shorthand notation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <base>:<arch>:
+
+Multi-base charms can also use shorthand notation when the platform builds on
+and builds for the same architecture. The platform name must be a
+:ref:`valid base <reference-bases>` and
+:ref:`debian architecture <reference-architectures>` formatted as
+``<base>:<arch>``.
+
+.. _reference-build-on:
+
+``build-on``
+~~~~~~~~~~~~
+
+``build-on`` is a list of architectures and optional bases that describes the
+environments where the platform can build. Each entry is formatted as
+``[<base>:]<arch>``.
+
+.. _reference-build-for:
+
+``build-for``
+~~~~~~~~~~~~~
+
+``build-for`` is a single-element list containing an architecture and optional
+base that describes the environment where the resulting charm can run. The
+entry is formatted as ``[<base>:]<arch>``.
+
+``build-for: [all]`` is a special keyword to denote an architecture-independent
+charm.
+
+.. _reference-architectures:
+
+Architectures
+-------------
+
+Charmcraft uses `Debian's naming convention`_ for architectures.
+
+The following architectures are supported:
+
+* amd64
+* arm64
+* armhf
+* i386
+* ppc64el
+* riscv64
+* s390x
+
+.. _reference-bases:
+
+Bases
+-----
+
+The ``base`` and ``build-base`` keywords determine the environments where the
+charm is built and where it is run.
+
+``base`` and ``build-base`` can't be defined for multi-base charms. Instead,
+the base is defined in the ``platforms`` keyword.
+
+``base``
+~~~~~~~~
+
+.. code-block:: yaml
+
+    base: <base>
+
+``base`` determines the runtime environment for the charm. It's formatted as
+``<distribution>@<series>`` where ``distribution`` is the name of a Linux
+distribution and ``series`` is the release series name.
+
+Supported bases are:
```

Comment:
This isn't actually true. Only Ubuntu 22.04+ are supported currently with the `base` keyword.

Also, while short term having this list here is fine, I think long-term I'd prefer to have a subcommand that tells us the supported bases.

Reply:
Agreed, and this is going to be outdated when 24.10 goes EOL in July 2025.  FWIW, dropping 24.10 will require manual intervention in craft-providers and charmcraft anyways.